lasagna-master/hemartin/lasagna_master.go

```diff
@@ -0,0 +1,70 @@
+package lasagna
+
+const noodleQuantityPerLayer int = 50
+const sauceQuantityPerLayer float64 = 0.2
+const originalQuantityPortions float64 = 2.0
+
+func PreparationTime(layers []string, layerPrepTime int) int {
+	if layerPrepTime <= 0 {
+		layerPrepTime = 2
+	}
+	return layerPrepTime * len(layers)
+
+}
+
+func Quantities(layers []string) (int, float64) {
+	return QuantitiesFast(layers)
+}
+
+// This was the first implementation
+func QuantitiesSlow(layers []string) (int, float64) {
+	count := map[string]int{
+		"noodles": 0,
+		"sauce":   0,
+	}
+
+	for _, layer := range layers {
+		if layer == "noodles" || layer == "sauce" {
+			count[layer]++
```

Comment:
This scales very badly. I would rewrite it as:
```
for _, layer := range layers {
	if _, ok := count[layer]; ok {
		count[layer]++
	}
}
```

Reply:
True, didn't thought about other types of layers.